template_example_test.go

```diff
@@ -207,147 +174,129 @@ func parseCode(_ context.Context, p *lexparse.Parser[*tmplNode]) error {
-	)
-
-	return nil
 }
 
 // parseVar handles replacement variables (e.g. the 'var' in {{ var }}).
 func parseVar(_ context.Context, p *lexparse.Parser[*tmplNode]) error {
-	fmt.Fprintf(os.Stderr, "parseVar: %#v\n", p.Pos().Value)
-
-	nameToken := p.Next()
+	switch token := p.Next(); token.Type {
+	case lexTypeIdentifier:
+		// Validate the variable name.
+		if !varNameRegexp.MatchString(token.Value) {
+			return tokenErr(errUnexpectedIdentifier, token)
+		}
 
-	// Validate the name token's type.
-	if nameToken.Type != lexTypeIdentifier {
-		return tokenErr(errUnexpectedIdentifier, nameToken)
-	}
+		// Add a variable node.
+		_ = p.Node(&tmplNode{
+			typ:     nodeTypeVar,
+			varName: token.Value,
+		})
 
-	// Validate the variable name.
-	if !varNameRegexp.MatchString(nameToken.Value) {
-		return tokenErr(errUnexpectedIdentifier, nameToken)
+		return nil
+	case lexparse.TokenTypeEOF:
+		return io.ErrUnexpectedEOF
+	default:
+		return tokenErr(errUnexpectedIdentifier, token)
 	}
-
-	// Add a variable node.
-	_ = p.Node(&tmplNode{
-		typ:     nodeTypeVar,
-		varName: nameToken.Value,
-	})
-
-	return nil
 }
 
 // parseVarEnd handles var end (e.g. '}}').
 func parseVarEnd(_ context.Context, p *lexparse.Parser[*tmplNode]) error {
-	fmt.Fprintf(os.Stderr, "parseVarEnd: %#v\n", p.Pos().Value)
-
-	// Validate the end variable.
-	token := p.Next()
-	// TODO(#95): Remove nil check.
-	if token == nil {
-		return tokenErr(errUnclosedVar, token)
-	}
-
-	// Validate the var end token type.
-	if token.Type != lexTypeVarEnd {
+	switch token := p.Next(); token.Type {
+	case lexTypeVarEnd:
+		// Go back to parsing code.
+		p.PushState(lexparse.ParseStateFn(parseCode))
+		return nil
+	case lexparse.TokenTypeEOF:
+		return io.ErrUnexpectedEOF
+	default:
 		return tokenErr(errUnexpectedIdentifier, token)
 	}
-
-	// Go back to parsing code.
-	p.PushState(lexparse.ParseStateFn(parseCode))
-	return nil
 }
 
 // parseBranch handles the start if conditional block.
 func parseBranch(_ context.Context, p *lexparse.Parser[*tmplNode]) error {
-	fmt.Fprintf(os.Stderr, "parseBranch: %#v\n", p.Pos().Value)
-
-	// Validate the if token
-	token := p.Next()
-	// TODO(#95): Remove nil check.
-	if token == nil {
-		return io.ErrUnexpectedEOF
-	}
-
-	if token.Type != lexTypeIdentifier || token.Value != tokenIf {
-		return tokenErr(errUnexpectedIdentifier, token)
-	}
+	switch token := p.Next(); token.Type {
+	case lexTypeIdentifier:
+		if token.Value != tokenIf {
+			return tokenErr(errUnexpectedIdentifier, token)
+		}
 
-	// Add a branch node.
-	_ = p.Push(&tmplNode{
-		typ: nodeTypeBranch,
-	})
+		// Add a branch node.
+		_ = p.Push(&tmplNode{
+			typ: nodeTypeBranch,
+		})
 
-	p.PushState(
-		// The first child will be the condition expression.
-		// Currently only a simple variable is supported.
-		lexparse.ParseStateFn(parseVar),
+		p.PushState(
+			// The first child will be the condition expression.
+			// Currently only a simple variable is supported.
```

Comment:
"Parse the condition expression. Currently only a simple variable is supported."